### ISSUE TITLE
SG inserts _removed and CBL can't handle it (some of the time) and tags it as illegal and kills the replicator

### ISSUE THREAD

**Issue by sellingsolutions** [couchbase/sync_gateway#1134]
I'm using the official Couchbase Lite 1.1.0

I've been trying to solve this for weeks, I'm probably going to write a proxy that removes the key "_removed" because it's wreaking complete havoc on my users.

The only clue I have as to why this is happening is that there are tons of conflicts, since my users work together on the same documents.

08:45:52.625‖ WARNING: CBL_Pusher[https://...........]: _bulk_docs got an error: {
    error = "bad_request";
    id = "customErrorCode_C06DC85B-42F4-4828-9D6E-44997C2A0CD9";
    reason = "user defined top level properties beginning with '_' are not allowed in document body";
    status = 400;
}

When I open that document up on my users iPad I notice that the json body has a _removed in the sqlite3 database in the revs table. For some reason CBL can't handle that as a document that "isGone" and just chokes the replicator instead.

![screen shot 2015-09-16 at 08 54 10](https://cloud.githubusercontent.com/assets/1778201/9898193/98da328a-5c50-11e5-96e4-680118836905.png)

**Comment by tleyden** [comment]
Which platform(s) are you running couchbase lite on?

**Comment by sellingsolutions** [comment]
"I'm using the official Couchbase Lite 1.1.0"

I've switched from Sync Gateway 1.1.0 to the latest master but it didn't do anything from what I can see.

**Comment by tleyden** [comment]
@sellingsolutions I meant the OS platform(s) you are running on.  Eg:
- Android native
- iOS native
- .NET / C#  (deployed to iOS or Android or both)
- Phonegap / Cordova (deployed to iOS or Android or both)

**Comment by sellingsolutions** [comment]
on the mobile side it's iOS (native) only

**Comment by tleyden** [comment]
@sellingsolutions Thanks!

Not sure this will strictly be needed by us, but anything you can provide in terms of instructions to reproduce or things that will help diagnose could be useful.  

When you say:

> chokes the replicator instead

Can you be more specific about what the behavior you are seeing with the replicator?  Also can you upload the logs from the iOS device into a github gist?  

Also, it would be useful to see your sync gateway config, in particular the sync function.

**Comment by sellingsolutions** [comment]
I sent a couple of example databases from my users to jens via e-mail (Dropbox links). 

The CBLReplicator usually dies when it gets a lot of errors, such as 401 or in this case warnings due to the illegal documents. The reason that I'm not running the CBL master is because I got 401 Unauthorized on everything and then the replicator dies immediately, hence why I switched back to CBL 1.1.0. but I'm still stuck with the problem that SG inserts _removed in a way that CBL can't handle.

My sg config file running Couchbase Sync Gateway/master(226819e) :
https://gist.github.com/sellingsolutions/5acee6aca21e490c0c66
